conda/common/io.py

```diff
@@ -343,16 +349,16 @@ def attach_stderr_handler(
 
 
 def timeout(timeout_secs, func, *args, default_return=None, **kwargs):
-    """Enforce a maximum time for a callable to complete.
-    Not yet implemented on Windows.
-    """
+    """Enforce a maximum time for a callable to complete."""
     if on_win:
-        # Why does Windows have to be so difficult all the time? Kind of gets old.
-        # Guess we'll bypass Windows timeouts for now.
-        try:
-            return func(*args, **kwargs)
-        except KeyboardInterrupt:  # pragma: no cover
-            return default_return
+        with ThreadLimitedThreadPoolExecutor(max_workers=1) as executor:
+            future = executor.submit(func, *args, **kwargs)
+            try:
+                return future.result(timeout=timeout_secs)
+            except KeyboardInterrupt:  # pragma: no cover
```

Comment:
I would be happy to write the test for this no-covered code

Reply:
Go ahead, not going to argue with that ;P

Reply:
I remember it's hard or barely-supported to kill threads? Will this do the trick?

Reply:
Following https://gist.github.com/clchiou/f2608cbe54403edb0b13 it should, but I haven't sat down to prove this with tests :-/